crates/c/src/binary.c

```diff
@@ -0,0 +1,209 @@
+#include "binary.h"
+#include <math.h>
+
+#if defined(__x86_64__)
+#include <immintrin.h>
+#endif
+
+#if defined(__x86_64__)
+
+#define WIDTH (512 / 8 / sizeof(size_t))
+
+__attribute__((target("arch=x86-64-v4,avx512vpopcntdq"))) extern float
```

Comment:
`avx512vpopcntdq` intrinsics are implemented in Rust so it is not necessary to write the code in C.

Reply:
completed in new commit